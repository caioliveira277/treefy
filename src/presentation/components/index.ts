export * from './button/button-component';
export * from './text-input/text-input-component';
export * from './link/link-component';
export * from './legend/legend-component';
export * from './picker-input/picker-input-component';
export * from './page-loading/page-loading-component';
export * from './toast-message/toast-message-component';
export * from './skeleton-loading/skeleton-loading-component';
export * from './empty-content/empty-content-component';
export * from './search-input/search-input-component';
export * from './animated-height/animated-height';
export * from './loading-overlay/loading-overlay-component';
