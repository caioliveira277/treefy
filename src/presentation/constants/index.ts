export const TREEFY_SUPPORT = {
  email: 'treefy.suporte@gmail.com',
  cel1: '(15) 99782-7880',
  cel2: '(11) 93203-7301',
};
