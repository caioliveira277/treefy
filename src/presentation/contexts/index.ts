export * from './authentication-context';
export * from './introduction-context';
export * from './toast-context';
