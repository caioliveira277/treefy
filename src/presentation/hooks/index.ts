export * from './use-debounce/use-debounce-hook';
