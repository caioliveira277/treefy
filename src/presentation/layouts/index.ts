export * from './public/public-layout';
export * from './profile/profile-layout';
