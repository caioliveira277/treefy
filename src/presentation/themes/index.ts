import { primaryTheme } from './primary-theme';
export const currentTheme = (() => primaryTheme)();
