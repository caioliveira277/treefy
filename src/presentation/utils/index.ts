export * from './get-icons/get-icons-utils';
export * from './format-date/format-date-utils';
