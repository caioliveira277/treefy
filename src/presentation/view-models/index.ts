export * from './authentication/authentication-view-model';
export * from './authentication/authentication-view-model-impl';

export * from './access/access-view-model';
export * from './access/access-view-model-impl';

export * from './email-confirmation/email-confirmation-view-model';
export * from './email-confirmation/email-confirmation-view-model-impl';

export * from './change-password/change-password-view-model';
export * from './change-password/change-password-view-model-impl';

export * from './signup/signup-view-model';
export * from './signup/signup-view-model-impl';

export * from './introduction/introduction-view-model';
export * from './introduction/introduction-view-model-impl';
