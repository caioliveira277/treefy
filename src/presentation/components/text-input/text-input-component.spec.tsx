import { mockRender } from '@/presentation/tests';
import {
  TextInputComponent,
  TextInputComponentProps,
} from './text-input-component';
import faker from '@faker-js/faker';
import { fireEvent, RenderAPI, waitFor } from '@testing-library/react-native';
import 'jest-styled-components/native';

type SutTypes = {
  sut: RenderAPI;
};
interface MakeSutProps extends Partial<TextInputComponentProps> {}

const makeSut = ({
  label = 'anyLabel',
  placeholderText = 'anyPlaceholder',
  iconName = 'mail',
  value = '',
  type,
  onChangeText = () => null,
}: MakeSutProps): SutTypes => {
  const sut = mockRender(
    <TextInputComponent
      label={label}
      type={type}
      placeholderText={placeholderText}
      iconName={iconName}
      value={value}
      onChangeText={onChangeText}
    />
  );
  return {
    sut,
  };
};

describe('TextInputComponent', () => {
  test('Should match with a snapshot', () => {
    const { sut } = makeSut({});
    const { toJSON } = sut;
    expect(toJSON()).toMatchSnapshot();
  });
  test('Should display text label', () => {
    const anyText = faker.datatype.string();
    const { sut } = makeSut({ label: anyText });
    const { getByTestId } = sut;
    const label = getByTestId('label');
    expect(label.children).toContain(anyText);
  });
  test('Should display text placeholder', () => {
    const anyText = faker.datatype.string();
    const { sut } = makeSut({ placeholderText: anyText });
    const { getByPlaceholderText } = sut;
    const input = getByPlaceholderText(anyText);
    expect(input).toBeTruthy();
  });
  test('Should contain the initial value', () => {
    const anyText = faker.datatype.string();
    const { sut } = makeSut({
      value: anyText,
    });
    const { getByDisplayValue } = sut;
    const input = getByDisplayValue(anyText);
    expect(input).toBeTruthy();
  });
  test('Should trigger the change text event', async () => {
    const handleChangeText = jest.fn();
    const anyText = faker.datatype.string();
    const { sut } = makeSut({
      onChangeText: handleChangeText,
    });
    const { getByTestId } = sut;
    const input = getByTestId('input');
    await fireEvent.changeText(input, anyText);
    expect(handleChangeText).toHaveBeenCalledWith(anyText);
  });
  test('Should find icon path by name', () => {
    const { sut } = makeSut({
      iconName: 'lock',
    });
    const { getByTestId } = sut;
    const genericIcon = getByTestId('genericIcon');
    expect(genericIcon.props.source).toEqual(1);
  });
  test('Should change password field visibility', async () => {
    const { sut } = makeSut({
      type: 'password',
    });
    const { getByTestId } = sut;
    const input = getByTestId('input');
    const visibilityPasswordButton = getByTestId('visibilityPasswordButton');
    expect(input.props.secureTextEntry).toBeTruthy();
    await waitFor(() => fireEvent.press(visibilityPasswordButton));
    expect(input.props.secureTextEntry).toBeFalsy();
  });
});
