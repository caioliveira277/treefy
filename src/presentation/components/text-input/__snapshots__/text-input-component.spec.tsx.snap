// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInputComponent Should match with a snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "width": "100%",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#253239",
          "fontFamily": "RobotoSlab_500Medium",
          "fontSize": 16,
          "marginBottom": 10,
        },
      ]
    }
    testID="label"
    theme={
      Object {
        "borders": Object {
          "border_radius_md": "10px",
          "border_radius_sm": "5px",
        },
        "colors": Object {
          "body": "#585960",
          "complete": "#4ABC86",
          "contrast": "#FCFDFF",
          "edit": "#4ABCBC",
          "placeholder": "#C6C6C6",
          "placeholder_light": "#EEEEEE",
          "primary": "#4ABC86",
          "rate_active": "#D56D6D",
          "rate_inactive": "#E5A6A7",
          "secondary": "#253239",
          "sun": "#FEF3A7",
          "water": "#B0D8FE",
          "white": "#ffffff",
        },
        "fonts": Object {
          "families": Object {
            "bold": "RobotoSlab_700Bold",
            "medium": "RobotoSlab_500Medium",
            "regular": "RobotoSlab_400Regular",
          },
          "sizes": Object {
            "lg": "16px",
            "md": "14px",
            "sm": "12px",
            "xl": "18px",
          },
        },
        "shadows": Object {
          "sm": Object {
            "elevation": 4,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.23,
            "shadowRadius": 2.62,
          },
        },
      }
    }
  >
    anyLabel
  </Text>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "borderColor": "#EEEEEE",
          "borderRadius": 5,
          "borderStyle": "solid",
          "borderWidth": 1.5,
          "flexDirection": "row",
          "minHeight": 45,
          "position": "relative",
          "width": "100%",
        },
      ]
    }
    theme={
      Object {
        "borders": Object {
          "border_radius_md": "10px",
          "border_radius_sm": "5px",
        },
        "colors": Object {
          "body": "#585960",
          "complete": "#4ABC86",
          "contrast": "#FCFDFF",
          "edit": "#4ABCBC",
          "placeholder": "#C6C6C6",
          "placeholder_light": "#EEEEEE",
          "primary": "#4ABC86",
          "rate_active": "#D56D6D",
          "rate_inactive": "#E5A6A7",
          "secondary": "#253239",
          "sun": "#FEF3A7",
          "water": "#B0D8FE",
          "white": "#ffffff",
        },
        "fonts": Object {
          "families": Object {
            "bold": "RobotoSlab_700Bold",
            "medium": "RobotoSlab_500Medium",
            "regular": "RobotoSlab_400Regular",
          },
          "sizes": Object {
            "lg": "16px",
            "md": "14px",
            "sm": "12px",
            "xl": "18px",
          },
        },
        "shadows": Object {
          "sm": Object {
            "elevation": 4,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.23,
            "shadowRadius": 2.62,
          },
        },
      }
    }
  >
    <Image
      resizeMode="center"
      source={1}
      style={
        Array [
          Object {
            "left": 10,
            "position": "absolute",
            "width": 24,
          },
        ]
      }
      testID="genericIcon"
    />
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="anyPlaceholder"
      placeholderTextColor="#C6C6C6"
      rejectResponderTermination={true}
      secureTextEntry={false}
      style={
        Array [
          Object {
            "color": "#253239",
            "fontFamily": "RobotoSlab_400Regular",
            "fontSize": 14,
            "paddingLeft": 38,
            "paddingRight": 42,
            "width": "100%",
          },
        ]
      }
      testID="input"
      theme={
        Object {
          "borders": Object {
            "border_radius_md": "10px",
            "border_radius_sm": "5px",
          },
          "colors": Object {
            "body": "#585960",
            "complete": "#4ABC86",
            "contrast": "#FCFDFF",
            "edit": "#4ABCBC",
            "placeholder": "#C6C6C6",
            "placeholder_light": "#EEEEEE",
            "primary": "#4ABC86",
            "rate_active": "#D56D6D",
            "rate_inactive": "#E5A6A7",
            "secondary": "#253239",
            "sun": "#FEF3A7",
            "water": "#B0D8FE",
            "white": "#ffffff",
          },
          "fonts": Object {
            "families": Object {
              "bold": "RobotoSlab_700Bold",
              "medium": "RobotoSlab_500Medium",
              "regular": "RobotoSlab_400Regular",
            },
            "sizes": Object {
              "lg": "16px",
              "md": "14px",
              "sm": "12px",
              "xl": "18px",
            },
          },
          "shadows": Object {
            "sm": Object {
              "elevation": 4,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.23,
              "shadowRadius": 2.62,
            },
          },
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
</View>
`;
